### DISCUSSION TITLE
500 VS Code failed to load.   Error: spawn /usr/lib/code-server/lib/node EAGAIN

### DISCUSSION THREAD

**Discussion by tokyode** [coder/code-server#1847] [Q&A]
500
VS Code failed to load.
Error: spawn /usr/lib/code-server/lib/node EAGAIN

Above error is shown on fresh install on　Debian GNU/Linux 10  following guide for install and accesing via ssh forwarding

config.yaml

bind-addr: 127.0.0.1:5020
auth: none
cert: false

installed as root user no other user exist.

Ssh forward command 

ssh -i kk -p5000 -N -L 5020:127.0.0.1:5020 root@serverIP

~# systemctl --user status code-server
● code-server.service - code-server
   Loaded: loaded (/usr/lib/systemd/user/code-server.servic
e; enabled; vendor preset: enabled)
   Active: active (running) since Sun 2020-06-28
 17:59:48 UTC; 11min ago
 Main PID: 106 (node)
    Tasks: 22 (limit: 22)
   Memory: 59.5M
   CGroup: /user.slice/user-0.slice/user@0.service/code-ser
ver.service
           ├─106 /usr/lib/code-server/lib/node /usr/lib/cod
e-server
           └─127 /usr/lib/code-server/lib/node /usr/lib/cod
e-server

Jun 28 17:59:48 lxcc802a3db systemd[100]: Starting code-ser
ver...
Jun 28 17:59:48 lxcc802a3db systemd[100]: Started code-serv
er.
Jun 28 17:59:49 lxcc802a3db code-server[106]: info  Using c
onfig file ~/.config/code-server/config.yaml
Jun 28 17:59:49 lxcc802a3db code-server[106]: info  Using u
ser-data-dir ~/.local/share/code-server
Jun 28 17:59:49 lxcc802a3db code-server[106]: info  c
Jun 28 17:59:49 lxcc802a3db code-server[106]: info  Using c
onfig file ~/.config/code-server/config.yaml
Jun 28 17:59:49 lxcc802a3db code-server[106]: info  Using u
ser-data-dir ~/.local/share/code-server
Jun 28 17:59:49 lxcc802a3db code-server[106]: info  code-se
rver 3.4.1 48f7c2724827e526eeaa6c2c151c520f48a61259
Jun 28 17:59:49 lxcc802a3db code-server[106]: info  HTTP se
rver listening on http://127.0.0.1:5020
Jun 28 17:59:49 lxcc802a3db code-server[106]: info    - No
authentication
Jun 28 17:59:49 lxcc802a3db code-server[106]: info    - Not
 serving HTTPS
Jun 28 18:00:18 lxcc802a3db code-server[106]: error spawn /
usr/lib/code-server/lib/node EAGAIN
Jun 28 18:01:51 lxcc802a3db code-server[106]: error spawn /
usr/lib/code-server/lib/node EAGAIN

**Comment by code-asher** [comment]
It seems this means you've hit a per-user process limit. You should be able to run `ulimit` to check the limit. You might also want to check how many processes you have with something like `ps aux`.

**Comment by tokyode** [comment]
> It seems this means you've hit a per-user process limit. You should be able to run `ulimit` to check the limit. You might also want to check how many processes you have with something like `ps aux`.

It doesn't look like that's the issue.

~# ulimit
unlimited
~# ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  1.9  21680  9640 ?        Ss   Jun28   0:00 /sbin/init
root          44  0.0  1.8  27552  9140 ?        Ss   Jun28   0:00 /lib/systemd/
root          59  0.0  1.4  19496  7168 ?        Ss   Jun28   0:00 /lib/systemd/
message+      60  0.0  0.8   8980  4028 ?        Ss   Jun28   0:00 /usr/bin/dbus
root          71  0.0  1.1   9484  5804 ?        Ss   Jun28   0:00 /sbin/dhclien
root          94  0.0  0.3   5380  1656 console  Ss+  Jun28   0:00 /sbin/agetty
root          95  0.0  1.3  15848  6848 ?        Ss   Jun28   0:00 /usr/sbin/ssh
root        2101  0.0  1.6  16948  8332 ?        Ss   03:11   0:00 sshd: root@pt
root        2104  0.0  1.7  21020  8920 ?        Ss   03:11   0:00 /lib/systemd/
root        2105  0.0  0.5  22660  2940 ?        S    03:11   0:00 (sd-pam)
root        2110  0.0  7.9 611660 39512 ?        Ssl  03:11   0:00 /usr/lib/code
root        2128  0.0  0.7   6988  3576 pts/0    Ss   03:11   0:00 -bash
root        2135  0.0  8.1 611660 40732 ?        Sl   03:11   0:00 /usr/lib/code
root        2146  0.0  0.6  10628  3112 pts/0    R+   03:12   0:00 ps aux

As you can see the server is fresh installed, not that many processes. ulimit is unlimited, vs-code-server is fresh install without using docker.

**Comment by code-asher** [comment]
Hmm it might be that `ulimit` doesn't give the full picture. I'm not super familiar with it unfortunately.

Here's the relevant section from the `fork` man page (https://www.man7.org/linux/man-pages/man2/fork.2.html)

```
       EAGAIN A system-imposed limit on the number of threads was
              encountered.  There are a number of limits that may trigger
              this error:

              *  the RLIMIT_NPROC soft resource limit (set via
                 setrlimit(2)), which limits the number of processes and
                 threads for a real user ID, was reached;

              *  the kernel's system-wide limit on the number of processes
                 and threads, /proc/sys/kernel/threads-max, was reached (see
                 proc(5));

              *  the maximum number of PIDs, /proc/sys/kernel/pid_max, was
                 reached (see proc(5)); or

              *  the PID limit (pids.max) imposed by the cgroup "process
                 number" (PIDs) controller was reached.

       EAGAIN The caller is operating under the SCHED_DEADLINE scheduling
              policy and does not have the reset-on-fork flag set.  See
              sched(7).
```

That makes it seem like the only way to get an `EAGAIN` is to hit some kind of limit. Is this a virtualized machine on a cloud provider? Maybe they've set a tiny limit of some kind?
